File: dotnet/Zirve.Sdk/Ingress/IngressManager.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using Zirve.Sdk.Config;

namespace Zirve.Sdk.Ingress;

public class IngressManager
{
    private readonly HttpClient _httpClient;
    private readonly string _url;

    public IngressManager(HttpClient httpClient, ConfigManager configManager)
    {
        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
        
        var cfg = configManager.Module("ingress");
        _url = (cfg.GetValueOrDefault("url", "http://traefik.kube-system.svc.cluster.local:8080") ?? "").TrimEnd('/');
    }

    public async Task<JsonElement> GetRoutesAsync()
    {
        try
        {
            var response = await _httpClient.GetAsync($"{_url}/api/http/routers");
            if (response.IsSuccessStatusCode)
            {
                using var doc = await JsonDocument.ParseAsync(await response.Content.ReadAsStreamAsync());
                return doc.RootElement.Clone();
            }
        }
        catch { }
        return default;
    }

    public async Task<JsonElement> GetMiddlewaresAsync()
    {
        try
        {
            var response = await _httpClient.GetAsync($"{_url}/api/http/middlewares");
            if (response.IsSuccessStatusCode)
            {
                using var doc = await JsonDocument.ParseAsync(await response.Content.ReadAsStreamAsync());
                return doc.RootElement.Clone();
            }
        }
        catch { }
        return default;
    }

    public async Task<bool> HealthAsync()
    {
        try
        {
            var response = await _httpClient.GetAsync($"{_url}/ping");
            return response.IsSuccessStatusCode;
        }
        catch
        {
            return false;
        }
    }
}

File: typescript/src/Ingress/IngressManager.ts
/**
 * Zirve Ingress Manager — Traefik
 *
 * Reads routing configurations and status from Traefik API.
 */
export class IngressManager {
  private config: Record<string, any>;
  private url: string;

  constructor(config: Record<string, any>) {
    this.config = config;
    this.url = (config.url || 'http://traefik.kube-system.svc.cluster.local:8080').replace(/\/$/, '');
  }

  public async routes(): Promise<any[]> {
    try {
      const response = await fetch(`${this.url}/api/http/routers`);
      return response.ok ? await response.json() : [];
    } catch {
      return [];
    }
  }

  public async middlewares(): Promise<any[]> {
    try {
      const response = await fetch(`${this.url}/api/http/middlewares`);
      return response.ok ? await response.json() : [];
    } catch {
      return [];
    }
  }

  public async health(): Promise<boolean> {
    try {
      const response = await fetch(`${this.url}/ping`);
      return response.ok;
    } catch {
      return false;
    }
  }
}
